### DISCUSSION TITLE
Use mitmproxy to dump communicated files

### DISCUSSION THREAD

**Discussion by ayhpw5j6** [mitmproxy/mitmproxy#5621] [Q&A]
Thanks for your great project. 
I checked [mitmdump](https://docs.mitmproxy.org/stable/#mitmdump) and apparently, it supports storing the communication in a single file but I would like the proxy to download all communicated files separately like this 
[https://blah.com/files/test.png](https://blah.com/files/test.png)
and test.png are somehow to be stored on the filesystem in a random/timestamped directory for future inspection.
How is this possible or is these tools not a great choice for this use case?

**Comment by Prinzhorn** [comment]
You can read the dump file and extract what you need: https://github.com/mitmproxy/mitmproxy/blob/a1631154d722ea4c82dcb72e0c84ad38c6991de5/examples/addons/io-read-saved-flows.py

Or write an addon that does what you want as you browse (e.g. use the `response` hook and write files with `flow.response.content` wherever you want) https://docs.mitmproxy.org/stable/api/events.html#HTTPEvents.response
